articles/Frontend/Architecture/coding-guide.md

```diff
@@ -284,54 +279,136 @@ const testTaker = Object.assign(aUser, assignee);
 The particularity of this pattern is the scope is shared between the target and the mixin : `this` will be shared.
 There are multiple ways to achieve this pattern, including using prototypes.
 
-*When to use it ?*
+_When to use it ?_
 When mixins are pure methods or stateless by preference. There shouldn't be any strong coupling between the target and the mixin (for example the mixin expects a property to be available in the target).
 
 3. Delegation
 
-_TBD_
+Delegation pattern is composition pattern where component delegates functionality to other module
+
+```js
+const person = {
+  name: "Carl",
+  allowanceLimit: 20,
+};
+
+const allowance = {
+  substract(amount) {
+    if (amount <= this.allowanceLimit) {
+      this.allowanceLimit -= amount;
+    }
+  },
+  getAllowanceLimit() {
+    return this.allowanceLimit;
+  },
+};
+
+function delegate(source, methods, provider) {
+  methods.forEach((methodName) => {
+    source[methodName] = function () {
+      return provider[methodName].apply(source, arguments);
+    };
```

Comment:
```suggestion
    source[methodName] = function (...args) {
      return provider[methodName].apply(source, args);
    };
```
I think today `arguments` is discouraged.

Reply:
**chore:** Totally agree. We should write the example in modern syntax. The `arguments` variable should be avoided. It is enforced by ESLint. I think we should add the ESLint implicit rules to this doc for better transparency and clarity.

**suggestion:** You could also use arrow function in the example, the lexical this of the delegate is not using it.
```suggestion
    source[methodName] = (...args) => provider[methodName].apply(source, args);
```

Reply:
Yes, I agree. Fixed that everywhere across the document